cpp/src/arrow/device.h

```diff
@@ -98,6 +101,71 @@ class ARROW_EXPORT Device : public std::enable_shared_from_this<Device>,
   /// \brief Return the DeviceAllocationType of this device
   virtual DeviceAllocationType device_type() const = 0;
 
+  /// \brief EXPERIMENTAL: An object that provides event/stream sync primitives
+  class ARROW_EXPORT SyncEvent {
+   public:
+    virtual ~SyncEvent() = default;
+
+    /// @brief Block until sync event is completed.
+    virtual Status wait() = 0;
+
+    /// @brief Make the provided stream wait on the sync event.
+    ///
+    /// Tells the provided stream that it should wait until the
+    /// synchronization event is completed without blocking the CPU.
+    /// @param stream Should be appropriate for the underlying device
+    virtual Status stream_wait(void* stream) = 0;
+
+    void set_stream(void* stream) {
```

Comment:
I'm not sure I understand the point of storing a stream in addition to the event. I think limiting this class to a single responsibility makes the API more easily understood.

Reply:
you're right, and @kkraus14 made the same suggestion so I'm removing this